dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -53,15 +59,68 @@ class TypedDataset[T](
     }
   }
 
-  /**
-    * Job returns an array that contains all the elements in this [[TypedDataset]].
+  /** Job returns an array that contains all the elements in this [[TypedDataset]].
     *
     * Running job requires moving all the data into the application's driver process, and
     * doing so on a very large [[TypedDataset]] can crash the driver process with OutOfMemoryError.
+    *
+    * Differs from `Dataset#collect` by wrapping it's result into a [[frameless.Job]].
+    */
+  def collect(): Job[Array[T]] =
```

Comment:
I'm not conviced `Vector` is a better option here. `Seq`s has no construction overhead, and fixes the equality issue on `Arrays`s:

    scala> val a: Seq[Int] = Array(1, 2, 3)
    a: Seq[Int] = WrappedArray(1, 2, 3)

    scala> val b: Seq[Int] = Array(1, 2, 3)
    b: Seq[Int] = WrappedArray(1, 2, 3)

    scala> a == b
    res0: Boolean = true

Of course you could always pattern match on a `Seq`, and if it's an `Array` do nasty things with it, but I would not consider this as a safety issue.